### ISSUE TITLE
LSOpenURLsWithRole() failed with error -10810 for the file

### ISSUE THREAD

**Issue by xanview** [mpv-player/mpv#5090]
Hi there, I followed the FAQ here https://github.com/mpv-player/mpv/wiki/FAQ - specifically "How can I make mpv the default application to open movie files on OSX?" - but I am getting an error (see below):

### mpv version and platform
mpv 0.27.0 (C) 2000-2017 mpv/MPlayer/mplayer2 projects
 built on Wed Nov  8 13:07:24 GMT 2017
macOS High Sierra (Version 10.13.1)

### Reproduction steps

```
brew remove mpv
brew install mpv --with-bundle
brew linkapps
brew install duti
duti -s io.mpv avi all
duti -s io.mpv mkv all
duti -s io.mpv mp4 all
open file.mkv
```
### Expected behavior

The file should play with mpv.

### Actual behavior

Shows an error:
```
$ open test.mkv
LSOpenURLsWithRole() failed with error -10810 for the file /Users/user/test.mkv.
```

Running mpv test.mkv works as expected however.

Any ideas what I did wrong?

**Comment by Akemi** [comment]
my guess the extension isn't properly associated with the App. try it with the conventional way "Get Info" on the file, choose the right App in "Open with" and click on "Change All".

**Comment by xanview** [comment]
I tried that :( - when I open the file, either by right click open or explicitly selecting mpv.app - nothing happens. I also tried clicking "change all".

![screen shot 2017-11-08 at 13 50 36](https://user-images.githubusercontent.com/578881/32552427-d68ec9ce-c48b-11e7-8d40-1161d0f14f95.jpg)
![screen shot 2017-11-08 at 13 50 30](https://user-images.githubusercontent.com/578881/32552428-d6a4b81a-c48b-11e7-929e-d25fcd3b456d.jpg)

**Comment by Akemi** [comment]
since you now updated your issue with the macOS version you are on i can tell you what's wrong. also how did you expect "open" to work when the mpv App doesn't even work for you?

this has been fixed with 77021cf6fe6a1813b832c8927c288e958b6f6845. it's in no release yet so you have to either wait or build from git master.
